aether-kernel/aether/kernel/api/project_artefacts.py

```diff
@@ -134,10 +136,16 @@ def upsert_project_artefacts(
     # 4. create/update the mappings
     for raw_mapping in mappings:
         ignore_fields = ['name']
-        if raw_mapping.get('definition') is None:
+        mapping_definition = raw_mapping.get('definition')
+        if mapping_definition is None:
             # in case of no mapping rules were indicated, do not update them.
             ignore_fields.append('definition')
-
+            mapping_definition = {'mappings': [], 'entities': {}}
+        else:
+            mapping_definition = {
+                'mapping': mapping_definition.get('mapping', []),
+                'entities': mapping_project_schemas,
```

Comment:
I found a future and very realistic problem here.

Example:

1. Pushing "Pipeline 1" with Entity Type "Person", the method creates the Schema "Person" with id 1.

2. Later, pushing another "Pipeline 2" with Entity Type "Person", the method create the Schema "Person_1" with id 2. In your mapping rules you have this:
```json
{
  "mapping: [
    ["zzz", "Person.id"],
  ],
  "entities": {
    "Person_1": 2,
  }
}
```

Reply:
Valid point. I am not sure if schema auto generated names should be a thing, since mapping rules depend on them. Unless the rules are to be programmatically changed on the fly. However, on the UI side of things this issue is taken care of by a preflight function which warns of existing objects with the same name and gives the user the options of updating the existing schema or change the name  for another schema to be created which in turn forces the user to modify the mapping rules.

Reply:
Wondering if this small change in the line `115` would make the trick

```python
# take the original given name to be used as entity entry in the related mapping instances
mapping_project_schemas[raw_schema.get('name', schema.name)] = str(project_schema.pk)
```
or maybe this one:
```python
# take the original given name to be used as entity entry in the related mapping instances
mapping_project_schemas[raw_schema.get('name', project_schema.name)] = str(project_schema.pk)
```